src/aiida_quantumespresso/workflows/ph/base.py

```diff
@@ -16,6 +17,12 @@
 PwCalculation = CalculationFactory('quantumespresso.pw')
 
 
+def validate_inputs(inputs):
+    """Validate the top level namespace."""
+    if 'qpoints_distance' not in inputs and 'qpoints' not in inputs:
+        return PhBaseWorkChain.exit_codes.ERROR_INVALID_INPUT_QPOINTS.message
```

Comment:
I would probably avoid to define the exit code, since it's a validation error, hence it's not a proper exit code.